pyccel/codegen/printing/ccode.py

```diff
@@ -800,9 +815,9 @@ def function_signature(self, expr):
         if not args:
             arg_code = 'void'
         else:
-            arg_code = ', '.join('{}'.format(self.function_signature(i))
-                        if isinstance(i, FunctionAddress) else '{0}{1}'.format(self.get_declare_type(i), i)
-                        for i in args)
+            arg_code = ', '.join('{}'.format(self.function_signature(i, False))
+                        if isinstance(i, FunctionAddress) else '{0}{1}'.format(self.get_declare_type(i), i.name)
+                        if print_arg_names else '{}'.format(self.get_declare_type(i)) for i in args)
```

Comment:
2 comments here:
1. Do you know where the space is added between the dtype and the name?
2. I wonder if it is more readable to do: `i.name if print_arg_names else ''` rather than having 2 format strings to reduce the size of this line?

Reply:
1 - space is add at get_declare_type

Reply:
:( I guess we'll just have to live with the space then